lstchain/calib/camera/pixel_threshold_estimation.py

```diff
@@ -49,30 +50,31 @@ def get_threshold_from_dl1_file(dl1_path, sigma_clean):
     picture_thresh: np.ndarray
         picture threshold calculated using interleaved pedestal events
     """
-    
+
     ped_mean_pe, ped_std_pe = get_bias_and_std(dl1_path)
 
     # If problem with interleaved pedestal std values occur, take pedestal
     # std values from calibration run.
     # Correct interleaved pedestal std array should have shape (2,2,1855)
     if ped_std_pe.shape[0] == 2:
-        interleaved_events_id = 1
+        interleaved_events_id = INTERLEAVED_CALIBRATION_ID
     else:
         interleaved_events_id = 0
     threshold_clean_pe = ped_mean_pe + sigma_clean*ped_std_pe
     # find pixels with std = 0 and mean = 0 <=> dead pixels in interleaved
     # pedestal event likely due to stars
-    unusable_pixels = get_unusable_pixels(dl1_path, interleaved_events_id)
+    unusable_pixels = get_unusable_pixels(dl1_path)
     # for dead pixels set max value of threshold
     threshold_clean_pe[interleaved_events_id, HIGH_GAIN, unusable_pixels] = \
         max(threshold_clean_pe[interleaved_events_id, HIGH_GAIN, :])
     # return pedestal interleaved threshold from data run for high gain
     return threshold_clean_pe[interleaved_events_id, HIGH_GAIN, :]
 
-def get_unusable_pixels(dl1_path, interleaved_events_id):
+def get_unusable_pixels(dl1_path):
     with tables.open_file(dl1_path) as f:
+        calibration_id = f.root.dl1.event.telescope.monitoring.pedestal.col('calibration_id')
         unusable_pixels = np.where(f.root[dl1_params_tel_mon_cal_key].col(
-                                   'unusable_pixels')[interleaved_events_id,
+                                   'unusable_pixels')[calibration_id[INTERLEAVED_CALIBRATION_ID],
```

Comment:
Why taking out the second argument? Should be called "pedestal_events_id" or something like that (can be interleaved or not; if not, by the way, there is no guarantee that the stars are in the right position). And then that should be used inside get_unusable_pixels where you now use INTERLEAVED_CALIBRATION_ID

Reply:
Done